src/azul/plugins/metadata/hca/contributor_matrices.py

```diff
@@ -0,0 +1,165 @@
+from collections import (
+    defaultdict,
+)
+from itertools import (
+    chain,
+    product,
+)
+
+from furl import (
+    furl,
+)
+
+from azul import (
+    config,
+)
+from azul.collections import (
+    NestedDict,
+)
+
+
+class ContributorMatrices:
+
+    @classmethod
+    def make_contributor_matrices(cls, catalog, files):
+        """
+        >>> from azul.doctests import assert_json
+        >>> def f(files):
+        ...     return assert_json(ContributorMatrices.make_contributor_matrices('dcp', files))
+
+        >>> f([{'uuid': 'u1', 'version': 'v1', 'name': 'n1', 'strata': 'a=1;b=2'}])
+        {
+            "a": {
+                "1": {
+                    "b": {
+                        "2": {
+                            "url": [
+                                "https://service.dev.singlecell.gi.ucsc.edu/fetch/repository/files/u1?version=v1&catalog=dcp"
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+
+        >>> f([{'uuid': 'u1', 'version': 'v1', 'name': 'n1', 'strata': 'a=1;b=2'},
+        ...    {'uuid': 'u2', 'version': 'v2', 'name': 'n2', 'strata': 'a=1;b=2'}])
+        {
+            "a": {
+                "1": {
+                    "b": {
+                        "2": {
+                            "url": [
+                                "https://service.dev.singlecell.gi.ucsc.edu/fetch/repository/files/u1?version=v1&catalog=dcp",
```

Comment:
The factory will also allow you to make these assertions shorter and independent of the deployment. I'm actually a bit surprised this test works on other deployments. Please find out why.

Reply:
This test does not work on other deployments. An oversight on my part, I had `develop` selected when testing as did Travis. Implementing suggested `file_url_func` parameter on the function and removing deployment specifics from doctest.